src/Microsoft.DotNet.ImageBuilder/src/Commands/BuildCommand.cs

```diff
@@ -30,6 +30,8 @@ public class BuildCommand : DockerRegistryCommand<BuildOptions, BuildOptionsBuil
         private readonly List<TagInfo> _processedTags = new List<TagInfo>();
         private readonly HashSet<PlatformData> _builtPlatforms = new();
 
+        protected IRegistryCredentialsProvider RegistryCredentialsProvider { get; init; }
```

Comment:
This doesn't have any derived classes so there's no need to make it `protected`. I would recommend following the same pattern as the other DI components. Same comment applies to this pattern in `PublishManifestCommand`.

```suggestion
        private readonly IRegistryCredentialsProvider _registryCredentialsProvider;
```

Reply:
Makes sense, I was on a fence making this a private field, but  thought it might be useful in the future - we could change it if needed later.

Reply:
Fixed with https://github.com/dotnet/docker-tools/pull/1342/commits/1bd3cfd6b0382fc3ac3b809e186aa794a2267f66